src/project/git/gitStatus.ts

```diff
@@ -0,0 +1,42 @@
+import { runCommand } from "../../action/cli/commandLine";
+
+export interface GitStatus {
+    isClean: boolean;
+    raw?: string;
+    sha: string;
+    branch: string;
+    upstream?: {
+        branch: string;
+        inSync: boolean;
+    };
+}
```

Comment:
this is an abstraction to get the git status. It aims to do it using porcelain commands, ones that are not going to change format in future versions of git.
It provides only the information that I needed; it doesn't attempt a full implementation of git status, which trust me is more complicated than you think, and is not necessary.